aws-api/src/main/java/com/amplifyframework/api/aws/utils/RestRequestFactory.java

```diff
@@ -125,12 +129,20 @@ public static Request createRequest(
     private static void populateBody(
             Request.Builder builder,
             byte[] data,
-            BodyCreationStrategy strategy) {
+            BodyCreationStrategy strategy
+    ) {
         if (data != null) {
             strategy.buildRequest(builder, data);
         }
     }
 
+    // Segment separator can be either '/' or '\'.
+    // HttpUrl.Builder assumes an empty URL if path segments
+    // begin with either character. Strip them before appending.
+    private static String stripLeadingSlashes(final String path) {
+        return path.replaceAll("^[\\\\/]+", "");
```

Comment:
Suggestion for readability (eschew regular expression):

```suggestion
        if (path.startsWith("/") || path.startsWith("\\")) {
            return path.substring(1);
        }
        
        return path;
```

Reply:
I know this is excessive, but it works for any combination of any number of slashes in the beginning. This way I *REALLY* ensure that the path doesn't begin with "/" or "\" or any combination of those two.

This way the method name actually makes sense, since it wants to completely strip the left side of any slashes.